apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/OperationBasedModelGroupBuilder.kt

```diff
@@ -174,16 +175,24 @@ internal class OperationBasedModelGroupBuilder(
       usedNames: MutableSet<String>,
       parentTypeConditions: List<String>,
   ): OperationField {
+    val rawTypeName = parentTypeConditions.lastOrNull() ?: ""
     if (selections.isEmpty()) {
       return OperationField(
           info = info,
           condition = condition,
           fieldSet = null,
+          shapes = emptyList(),
+          rawTypeName = rawTypeName
       )
     }
 
     val selfPath = path + "." + info.responseName
-
+    val shapes = shapes(
+        schema,
+        allFragmentDefinitions,
+        selections.map { it.selection }.filter { it !is GQLFragmentSpread },
+        rawTypeName
+    ).filter { it.possibleTypes.isNotEmpty() }
```

Comment:
Out of curiosity, is that filtering anything? By definition, I would expect `shapes()` to only return shapes with non -empty possibleTypes

Reply:
Yes, it returns empty possibleTypes for `something` field below:

```graphql
query TestOperation {
  something {
    id
    ... on Human {
      name
    }
    ... on ThingWithName {
      name
    }
  }
}

# schema
type Query {
  something: ThingWithID
}
interface ThingWithID {
  id: ID!
}

interface ThingWithName {
  name: String!
}
type Human implements ThingWithID & ThingWithName {
  id: ID!
  name: String!
}
```
`shapes()` will return result below:
```
0 = {Shape@2727} Shape(typeSet=[ThingWithID], possibleTypes=[Droid])
1 = {Shape@2728} Shape(typeSet=[ThingWithID, Human], possibleTypes=[])
2 = {Shape@2729} Shape(typeSet=[ThingWithID, ThingWithName], possibleTypes=[Pet])
3 = {Shape@2730} Shape(typeSet=[ThingWithID, Human, ThingWithName], possibleTypes=[Human])
```

Reply:
Right. That's for interfaces in responseBased codegen (https://github.com/apollographql/apollo-kotlin/blob/071039c17dd42b21ecbfe081f5d1b69363407fbd/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/ResponseBasedModelGroupBuilder.kt#L409)

We might want to add a parameter to avoid returning those.

Reply:
Should I add a parameter now?